### PR TITLE
Recipe for Python part of Apache Arrow

### PR DESCRIPTION
**Pull request by wesm**
Requires #1729 and #992

### CODE REVIEW

File: recipes/pyarrow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{% set version = "0.1.pre" %}
+{% set git_sha = "2d8e82056afdcf125e6e512f96007389ce79c1c7" %}
+
+package:
+  name: pyarrow
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/apache/arrow.git
```
Comment (chrisaycock) [inline recipes/pyarrow/meta.yaml]: Does the feedback from @ocefpaf on arrow-cpp [here](https://github.com/conda-forge/staged-recipes/pull/1729#pullrequestreview-3494413) also apply?

Comment (ocefpaf) [inline recipes/pyarrow/meta.yaml]: Yes. If you could make those changes it will make this clearer.

**Comment by chrisaycock** [comment]
The build and dependencies LGTM.

File: recipes/pyarrow/meta.yaml (hunk 1 above)
Comment (wesm) [inline recipes/pyarrow/meta.yaml]: done, will wait for #992 to become feedstock then get this to a green build

**Comment by wesm** [comment]
This is failing because of PARQUET-747. I will update the parquet-cpp build and report back when this build is green

**Comment by chrisaycock** [comment]
Link: https://issues.apache.org/jira/browse/PARQUET-747

**Comment by wesm** [comment]
OK, the parquet-cpp conda-forge package has been updated, looks like we're good to go whenever the OS X build passes

File: recipes/pyarrow/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+source:
+  fn: {{ commit }}.tar.gz
+  url: https://github.com/apache/arrow/archive/{{ commit }}.tar.gz
+  sha256: 01ff68b9ab2785cc2a465fedc62536faab4ff461b6d42b9b00df15bb2e41dfa5
+
+build:
+  number: 0
+  rpaths:
+    - lib                                                        # [unix]
+    - lib/python{{environ.get('PY_VER')}}/site-packages/pyarrow  # [unix]
```
Comment (ocefpaf) [inline recipes/pyarrow/meta.yaml]: The linter is failing here. Would `{{ environ['PY_VER'] + '*' }}` work instead? That is known to pass.

Comment (wesm) [inline recipes/pyarrow/meta.yaml]: done

**Comment by wesm** [comment]
@ocefpaf nope, that broke my rpath. trying again without the `'*'`

**Comment by ocefpaf** [comment]
> @ocefpaf nope, that broke my rpath. trying again without the `'*'`

No problem. We should probably fix the linter and not the recipe anyways.

**Comment by wesm** [comment]
This is good to go once the build is green

**Comment by wesm** [comment]
Argh, well we have an rpath issue on OS X now. Looking.

**Comment by wesm** [comment]
@ocefpaf I'm pretty bewildered by the OS X rpath stuff, do you have any advice? Our package builds on our Travis CI are working for OS X, e.g. https://travis-ci.org/apache/arrow/jobs/166942076

The issue here is that there is a C++ shared library nested under `pyarrow/` and this is being hidden (?) by the install_name_tool. I can't tell what's different about the conda forge environment that would make this fail. It's XCode 6.0 versus 6.1, but not sure if that's enough to screw things up.

**Comment by ocefpaf** [comment]
> @ocefpaf I'm pretty bewildered by the OS X rpath stuff, do you have any advice?

Sorry, never used a Mac in my life. Maybe @msarahan can help here.

> Our package builds on our Travis CI are working for OS X, e.g. https://travis-ci.org/apache/arrow/jobs/166942076

Are you building with conda there? Note that the [xcode](https://github.com/apache/arrow/blob/master/.travis.yml#L55) is different too, but I don't think that is the issue.

> The issue here is that there is a C++ shared library nested under `pyarrow/` and this is being hidden (?) by the install_name_tool. I can't tell what's different about the conda forge environment that would make this fail. It's XCode 6.0 versus 6.1, but not sure if that's enough to screw things up. 

I maintain [this](https://github.com/conda-forge/esmf-feedstock/blob/master/recipe/build.sh#L24-L29)  recipe where we had to manually change paths with `install_name_tool`. Maybe that is the case here?

**Comment by wesm** [comment]
I'm trying to resolve this in apache/arrow#171, setting this to build from that PR

**Comment by ocefpaf** [comment]
@wesm `Linux` is OK and you are already skipping Windows. Do you want to skip `OS X` for now, get this into a feedstock, and experiment there? It would make your tests more agile and you would free `AppVeyor`'s cue. (Even when skipping AppVeyor will start. You can force skip it by adding `skip appveyor` in the commit message though.)

**Comment by wesm** [comment]
Let me see if this commit fixed it, and if not I'll make this Linux only to go to feedstock

**Comment by wesm** [comment]
@msarahan @ocefpaf @jakirkham this problem appears to be localized to the old version of XCode being used in conda-forge -- I left a comment on the bigger discussion about OS X architecture

**Comment by wesm** [comment]
@ocefpaf I disabled all but Linux for now, I can re-enable OS X when the XCode toolchain issues are resolved. thanks!

**Comment by ocefpaf** [comment]
Linux is no longer  :unamused: 

```shell
import: 'pyarrow'
Traceback (most recent call last):
  File "/staged-recipes/build_artefacts/test-tmp_dir/run_test.py", line 26, in <module>
    import pyarrow
  File "/opt/conda/envs/_test/lib/python3.5/site-packages/pyarrow/__init__.py", line 20, in <module>
    import pyarrow.config
ImportError: /opt/conda/envs/_test/lib/python3.5/site-packages/pyarrow/../../../libparquet_arrow.so: undefined symbol: _ZN5arrow16PrimitiveBuilderINS_11BooleanTypeEE6FinishEv
TESTS FAILED: pyarrow-0.1.post-0
```

**Comment by wesm** [comment]
Visibility is such a pain! I will fix

**Comment by ocefpaf** [comment]
>  I can re-enable OS X when the XCode

@wesm this kind of changes needs more discussion. In principle I am :+1: to that but we need to consensus as this change should happen everywhere. (In fact, for my local builds, I use homebrew's `gcc` so I am not tied to `OS X` specific stuff.)

It would be nice if you, or someone in your group, could participate some of our meetings and request that item to be discussed. See https://conda-forge.hackpad.com/conda-forge-meetings-2YkV96cvxPG

Maybe we could start by putting together a conda-forge-enhancement-proposal to move to a modern XCode. See  https://github.com/conda-forge/conda-forge-enhancement-proposals

**Comment by wesm** [comment]
Oh, actually this is because the parquet-cpp package is out of date, whoops.

**Comment by wesm** [comment]
@ocefpaf looks like something is broken in the conda forge toolchain now?

**Comment by ocefpaf** [comment]
> @ocefpaf looks like something is broken in the conda forge toolchain now?

Good old "turn it off and on again" seems to fix it :smile:

I re-started `CircleCI` and cancelled `AppVeyor` and `Travis-CI` since you are only building `Linux` for now.

**Comment by wesm** [comment]
thank you!

**Pull request merged**